src/server/test/routes/logsRouteTests.js

```diff
@@ -0,0 +1,302 @@
+           );
+           
+           // Should only contain WARN log
+           expect(warnLogs).to.have.lengthOf(1);
+           expect(warnLogs[0].logType).to.equal('WARN');
+           expect(warnLogs[0].logMessage).to.equal(warnMessage);
+       });
+
+       mocha.it('should filter logs by multiple types', async () => {
+           const infoMessage = 'Test INFO multi-type';
+           const warnMessage = 'Test WARN multi-type';
+           const errorMessage = 'Test ERROR multi-type';
+
+           // Insert different types of logs
+           await chai.request(app)
+               .post('/api/logs/info')
+               .set('token', token)
+               .send({ message: infoMessage });
+
+           await chai.request(app)
+               .post('/api/logs/warn')
+               .set('token', token)
+               .send({ message: warnMessage });
+
+           await chai.request(app)
+               .post('/api/logs/error')
+               .set('token', token)
+               .send({ message: errorMessage });
+
+           // Query for INFO and WARN logs
+           const response = await chai.request(app)
+               .get('/api/logs/logsmsg/getLogsByDateRangeAndType')
+               .set('token', token)
+               .query({ 
+                   timeInterval: moment().subtract(1, 'minute').toISOString(),
+                   logTypes: 'INFO,WARN',
+                   logLimit: '100'
+               });
+
+           expect(response.status).to.equal(200);
+           const testLogs = response.body.filter(log => 
+               log.logMessage === infoMessage || log.logMessage === warnMessage || log.logMessage === errorMessage
+           );
+           
+           // Should contain INFO and WARN logs, but not ERROR
+           expect(testLogs).to.have.lengthOf(2);
+           const logTypes = testLogs.map(log => log.logType).sort();
+           expect(logTypes).to.deep.equal(['INFO', 'WARN']);
+       });
+
+       mocha.it('should respect log limit parameter', async () => {
+           const messagePrefix = 'Limit test message';
+
+           // Insert multiple logs
+           for (let i = 0; i < 5; i++) {
+               await chai.request(app)
+                   .post('/api/logs/info')
+                   .set('token', token)
+                   .send({ message: `${messagePrefix} ${i}` });
+           }
+
+           // Query with limit of 2
+           const response = await chai.request(app)
+               .get('/api/logs/logsmsg/getLogsByDateRangeAndType')
+               .set('token', token)
+               .query({ 
+                   timeInterval: moment().subtract(1, 'minute').toISOString(),
+                   logTypes: 'INFO',
+                   logLimit: '2'
+               });
+
+           expect(response.status).to.equal(200);
+           expect(response.body).to.be.an('array');
+           
+           // Should not return more than the limit
+           expect(response.body.length).to.be.at.most(2);
```

Comment:
I think in this case it should be exactly 2 since you inserted 5. If it was less than 2 then it made a mistake.